From 21b871c64b756149c042ee2967ac755b08c92690 Mon Sep 17 00:00:00 2001
From: ywan171 <yi.a.wang@intel.com>
Date: Tue, 22 Jan 2013 22:13:22 +0800
Subject: [PATCH 03/56] [PATCH] [PORT FROM MAIN]libstagefright: If nativewindow
 equals null, do not set scale mode

if nativewindow == NULL,native_window_set_scaling_mode will cause
TOMBSTONE error, add protection before native_window_set_scaling_mode

Change-Id: Icd94eaa129ff65f77a1bac9a17af18aa494b269e
Category: aosp improvement
Domain: Video.Media-streaming
Origin: internal
Signed-off-by: ywan171 <yi.a.wang@intel.com>
Signed-off-by: chuzhoux <chux.zhou@intel.com>
---
 media/libstagefright/ACodec.cpp | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/media/libstagefright/ACodec.cpp b/media/libstagefright/ACodec.cpp
index 7e31dfb..ee5341a 100755
--- a/media/libstagefright/ACodec.cpp
+++ b/media/libstagefright/ACodec.cpp
@@ -4115,10 +4115,11 @@ bool ACodec::LoadedState::onConfigureComponent(
                 static_cast<NativeWindowWrapper *>(obj.get()));
         CHECK(nativeWindow != NULL);
         mCodec->mNativeWindow = nativeWindow->getNativeWindow();
-
-        native_window_set_scaling_mode(
-                mCodec->mNativeWindow.get(),
-                NATIVE_WINDOW_SCALING_MODE_SCALE_TO_WINDOW);
+        if (mCodec->mNativeWindow != NULL) {
+            native_window_set_scaling_mode(
+                    mCodec->mNativeWindow.get(),
+                    NATIVE_WINDOW_SCALING_MODE_SCALE_TO_WINDOW);
+        }
     }
     CHECK_EQ((status_t)OK, mCodec->initNativeWindow());
 
-- 
2.7.4

